CMakeLists.txt

```diff
@@ -168,6 +170,11 @@ IF(${BUILD_OPENCL})
     ADD_SUBDIRECTORY(src/backend/opencl)
 ENDIF()
 
+IF(${BUILD_HETEROGENOUS_API})
+    ADD_SUBDIRECTORY(src/api/hapi)
+    ADD_SUBDIRECTORY(hapi_examples)
```

Comment:
I don't think the examples need to exist in different directories. We can add dummy variations of all hapi specific functions in each backend.